src/infra/ChatAnalyticsReporter.ts

```diff
@@ -0,0 +1,74 @@
+import type { EventPayload } from '../models/chat';
+import type ChatAnalyticsConfig from '../models/config/ChatAnalyticsConfig';
+
+interface EventsAPIResponse {
+  id: string;
+}
+
+/**
+ * A class to report chat analytics data. Uses the provided API key, environment,
+ * and partition to determine the endpoint for reporting events.
+ */
+export default class ChatAnalyticsReporter {
```

Comment:
nit: we should mark this as public since it's being returned in our exported `provideChatAnalytics` function

Reply:
Yep good call.